File: src/ResponseType.cs

using System.Security.Cryptography;

public enum CipherMethod
{
  Encrypt,
  Decrypt,
}

public enum CipherAlgo
{
  AES,
}

public record MyResponseType(
  CipherMethod CipherMethod,
  CipherAlgo CipherAlgo,
  CipherMode CipherMode,
  int CipherLength,
  string Key,
  string? IV,
  string Data,
  string? Encrypted,
  string? Decrypted
);

File: web/src/ResponseType.ts

enum MyCipherMethod {
  Encrypt,
  Decrypt,
}

enum MyCipherAlgo {
  AES
}

enum MyCipherMode {
  ECB
}

interface MyResponseType {
  cipherMethod: MyCipherMethod;
  cipherAlgo: MyCipherAlgo;
  cipherMode: MyCipherMode;
  cipherLength: number;
  key: string;
  iv?: string;
  data: string;
  encrypted?: string;
  decrypted?: string;
}

export type { MyResponseType, MyCipherMethod, MyCipherAlgo, MyCipherMode };
